ospf6d/ospf6_top.c

```diff
@@ -731,8 +726,9 @@ DEFUN (no_ospf6_interface_area,
 	}
 
 	/* parse Area-ID */
-	if (inet_pton(AF_INET, argv[idx_ipv4]->arg, &area_id) != 1) {
-		vty_out(vty, "Invalid Area-ID: %s\n", argv[idx_ipv4]->arg);
+	area_id = htonl(strtoul(str, &ep, 10));
```

Comment:
`ep` is unnecessary, the CLI core ensures you get either a v4 or a number in the specified range. You can also drop the error check.

Reply:
I needed to keep the `strtoul()` and `inet_pton()` calls, but have swapped them around and removed `ep`.